velox/expression/CoalesceExpr.cpp

```diff
@@ -58,6 +58,11 @@ void CoalesceExpr::evalSpecialForm(
 
     if (!result->mayHaveNulls()) {
       // No nulls left.
+      return;
+    }
+
+    if (context.errors()) {
+      context.deselectErrors(*activeRows);
     }
 
     decodedVector.get()->decode(*result, *activeRows);
```

Comment:
Do we need to check that activeRows is not empty after context.deselectErrors ? I don't remember if decode works with empty rows.

Reply:
I think DecodedVector::decode() works with empty rows. There are unit tests `TEST_F(DecodedVectorTest, emptyRowsDictOverConstWithNulls)` and `TEST_F(DecodedVectorTest, emptyRowsMultiDict)`.